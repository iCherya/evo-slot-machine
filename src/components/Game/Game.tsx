import React from 'react';

export const Game: React.FC = () => {
  return <div>Game component</div>;
};
