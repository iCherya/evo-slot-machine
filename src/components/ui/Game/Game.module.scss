@import '@/styles/variables.scss';

.wrapper {
  height: 300px; // for now - will be replaced with dynamic height
  padding: $spacing-base;
  position: relative;
}

.controls {
  position: absolute;
  bottom: 1em;
  left: 50%;
  transform: translateX(-50%);
  display: flex;
  justify-content: center;
  gap: 1em;
}
