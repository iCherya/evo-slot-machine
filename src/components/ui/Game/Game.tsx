import React from 'react';
import { observer } from 'mobx-react-lite';

import { useStore } from '@/store';
import { GameControls } from '@/components/logic/GameControls';
import { SlotMachine } from '@/components/logic/SlotMachine';

import styles from './Game.module.scss';

export const Game: React.FC = observer(() => {
  const { game } = useStore();

  return (
    <>
      <SlotMachine />
      <GameControls />
      {game.showWinAnimation && (
        <div className={styles.banner} data-testid="banner">
          +{game.winCount}
        </div>
      )}
    </>
  );
});
