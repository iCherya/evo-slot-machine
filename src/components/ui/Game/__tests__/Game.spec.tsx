import { render } from '@testing-library/react';

import { Game } from '@/components/ui/Game';

jest.mock('@/components/logic/SlotMachine', () => ({
  SlotMachine: () => <div data-testid="slot-machine" />,
}));

jest.mock('@/components/logic/GameControls', () => ({
  GameControls: () => <div data-testid="game-controls" />,
}));

describe('Game', () => {
  it('should render SlotMachine and GameControls components', () => {
    const { container, getAllByTestId } = render(<Game />);

    expect(getAllByTestId('slot-machine').length).toBe(1);
    expect(getAllByTestId('game-controls').length).toBe(1);

    expect(container).toMatchSnapshot();
  });
});
