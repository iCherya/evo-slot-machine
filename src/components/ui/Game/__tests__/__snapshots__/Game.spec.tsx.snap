// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render SlotMachine and GameControls components 1`] = `
<div>
  <div
    data-testid="slot-machine"
  />
  <div
    data-testid="game-controls"
  />
</div>
`;
